File: Back-restaurante/src/entities/DetallePurchaseOrder.js
const { EntitySchema } = require("typeorm");

module.exports = new EntitySchema({
  name: "DetallePurchaseOrder",
  tableName: "detalle_purchase_order",
  columns: {
    id_detalle_po: {
      primary: true,
      type: "number",
      generated: true,
    },
    id_po_fk: {
      type: "number",
      nullable: false,
    },
    id_ingrediente_fk: {
      type: "number",
      nullable: false,
    },
    precio: {
      type: "number",
      precision: 10,
      scale: 2,
      nullable: true,
    },
    cantidad: {
      type: "number",
      nullable: true,
    },
  },
  relations: {
    purchaseOrder: {
      type: "many-to-one",
      target: "PurchaseOrder",
      joinColumn: {
        name: "id_po_fk",
        referencedColumnName: "id_po"
      }
    },
    ingrediente: {
      type: "many-to-one",
      target: "StoreIngrediente",
      joinColumn: {
        name: "id_ingrediente_fk",
        referencedColumnName: "id_ingrediente"
      }
    }
  }
});

File: backend/backend/src/main/java/restaurante/backend/DetallePurchaseOrder/domain/DetallePurchaseOrder.java
package restaurante.backend.DetallePurchaseOrder.domain;

import jakarta.persistence.*;
import lombok.Data;
import restaurante.backend.StoreIngrediente.domain.StoreIngrediente;
import restaurante.backend.PurchaseOrder.domain.PurchaseOrder;
@Entity
@Table(name = "detalle_purchase_order")
@Data
public class DetallePurchaseOrder {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id_detalle_po;

    @ManyToOne
    @JoinColumn(name = "id_po_fk", nullable = false)
    private PurchaseOrder ordenCompra;

    @ManyToOne
    @JoinColumn(name = "id_ingrediente_fk", nullable = false)
    private StoreIngrediente ingrediente;

    private double precio;
    private int cantidad;
}
